### DISCUSSION TITLE
Looking into Target for STM411CEU6 with sensors and source for a development board

### DISCUSSION THREAD

**Discussion by synersignart** [iNavFlight/inav#8272] [Ideas]
are their possibility to 

#define MPU6000_I2C_BUS         
#define MPU6500_I2C_BUS     
#define Gy-bmi160_I2C_BUS     

these sensors as I2C as I only see them as ISP in Targets

would like to know if these is possible . 
the reason im looking to start from development board is to understand how to compile inav in its simplest form yet to seen a release of the source code for this type of board

for Development boards
STM411CEU6 BlackPill 
STM32f746 discovery
STM32F407VET6
STM32F446

**Comment by stronnag** [comment]
Maybe this  [wiki article](https://github.com/iNavFlight/inav/wiki/Building-custom-firmware) will get you started.

**Reply by synersignart** [reply]
ive read thru it also found a few boards that has the STM32F411CEU6 but I cant get it to connect after the DFU flash suggesting the Target is not compatible or im missing something . 
Matek 411 is the closest match but this causes some errors
target_stm32f411xe

PS 
im completely new to INAV so I may need some one to help me do a walk through or a frame work to get started . 

Riza has release an Hex but yet to release the Target folder source and all the information that was done to achieve this but id ask if anyone else here also working on development board as I see i may need to do modification if I could understand how the process is done . 
https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware

---------------------------------------------------------------------
@stronnag I Appreciate your updates on the MWTOOLS this really helps for our community using Linux, cheers :)

**Comment by synersignart** [comment]
some update with some problems as I cant find an suitable board target everytime I load the Hex the board wont boot up 
still needed help is sorting the target issue

![image](https://user-images.githubusercontent.com/22503442/182334260-48d62d48-d2f8-4f04-951d-c09397eba9cb.png)
![viber_image_2022-08-02_16-41-29-472](https://user-images.githubusercontent.com/22503442/182334361-47d5d92e-944a-478d-aeac-775f041349d1.jpg)

**Comment by erstec** [comment]
> are their possibility to
> 
> #define MPU6000_I2C_BUS #define MPU6500_I2C_BUS #define Gy-bmi160_I2C_BUS
> 
> these sensors as I2C as I only see them as ISP in Targets
> 
> would like to know if these is possible . the reason im looking to start from development board is to understand how to compile inav in its simplest form yet to seen a release of the source code for this type of board
> 
> for Development boards STM411CEU6 BlackPill STM32f746 discovery STM32F407VET6 STM32F446

@synersignart I believe we found "solution" and "way to do it" on Discord already.
